CMakeLists.txt

```diff
@@ -0,0 +1,297 @@
+cmake_minimum_required(VERSION 3.1)
+
+if(CMAKE_VERSION VERSION_GREATER 3.14)
+  # MSVC runtime library flags are selected by the CMAKE_MSVC_RUNTIME_LIBRARY abstraction.
+  cmake_policy(SET CMP0091 NEW)
+  # MSVC warning flags are not in CMAKE_<LANG>_FLAGS by default.
+  cmake_policy(SET CMP0092 NEW)
+endif()
+
+# The package (a.k.a. release) version is based on semantic versioning 2.0.0 of
+# the API. All changes in experimental modules are treated as
+# backwards-compatible and therefore at most increase the minor version.
+project(secp256k1 VERSION 0.2.1 LANGUAGES NONE)
+
+# The library version is based on libtool versioning of the ABI. The set of
+# rules for updating the version can be found here:
+# https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+# All changes in experimental modules are treated as if they don't affect the
+# interface and therefore only increase the revision.
+set(${PROJECT_NAME}_LIB_VERSION_CURRENT 1)
+set(${PROJECT_NAME}_LIB_VERSION_REVISION 1)
+set(${PROJECT_NAME}_LIB_VERSION_AGE 0)
+
+enable_language(C)
+set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_EXTENSIONS OFF)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Agree. Given the build targets, this seems like one of the teeny tiny subset of projects that may actually care. Like @real-or-random says here: https://github.com/bitcoin-core/secp256k1/pull/1113#discussion_r1124273356 , I think we should probably skip this for now and discuss it in a (likely one of many of such) follow-up PR.

FWIW downstream Core sets this, so there's some precedent at least: https://github.com/bitcoin/bitcoin/blob/master/configure.ac#L1992

Reply:
@theuni (Ignoring that this discussion may be long to a separate PR, what's the stories of enabling PIC for non-shared libraries? Are there any advantages or disadvantages? Why does Core set this for libsecp256k1? (This goes back to 2014 https://github.com/bitcoin/bitcoin/commit/07a99017033b23f840f602d768efa87e0e914e90 ) Should we enable this in our test and example binaries? (I guess that doesn't matter at all.)

Reply:
@real-or-random Core builds a libsecp and links it into a shared libconsensus/libkernel, as well as hardened (-pie) binaries. For that reason, we always want a pic libsecp, even though we're building it statically.

For some platforms (x86 for ex) there may be a significant overhead, but we accept it because of the hardening (aslr) that comes with it. Some projects on niche and/or resource-starved architectures may not be so willing to accept the overhead.

Reply:
Just revisiting this to confirm...

From `POSITION_INDEPENDENT_CODE`'s docs

> This property is True by default for SHARED and MODULE library targets and False otherwise. This property is initialized by the value of the [CMAKE_POSITION_INDEPENDENT_CODE](https://cmake.org/cmake/help/latest/variable/CMAKE_POSITION_INDEPENDENT_CODE.html#variable:CMAKE_POSITION_INDEPENDENT_CODE) variable if it is set when a target is created.


This matches the current libtool behavior, so leaving it alone seems correct to me. Core should be able to set it at the top level in order to get PIC in our static libsecp.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/secp256k1/pull/1113#issuecomment-1457095241).